mcs/class/corlib/corert/Decimal.DecCalc.cs

```diff
@@ -703,11 +705,7 @@ private static unsafe uint DivByConst(uint* result, uint hiRes, out uint quotien
 #else
                     // 32-bit RyuJIT doesn't convert 64-bit division by constant into multiplication by reciprocal. Do half-width divisions instead.
                     Debug.Assert(power <= ushort.MaxValue);
-#if BIGENDIAN
-                    const int low16 = 2, high16 = 0;
-#else
-                    const int low16 = 0, high16 = 2;
-#endif
+                    const int low16 = BitConverter.IsLittleEndian ? 0 : 2, high16 = BitConverter.IsLittleEndian ? 2 : 0;
```

Comment:
hm.. does it compile? 😨

Reply:
it seemed to for me, but I suspect that may be because for all I know, BIT64 is defined

Reply:
it's const (it cannot be initialized conditionally)

Reply:
compiles for me inside Decimal too, kind of magic

Reply:
this doesn't compile when extracted; i'll de-const it, but I suspect Core might not like it being non-const

```
const.cs(5,49): error CS0133: The expression being assigned to `low16' must be a constant or default value
const.cs(6,50): error CS0133: The expression being assigned to `high16' must be a constant or default value
Compilation failed: 2 error(s), 0 warnings
```

```csharp
using System;

class c {
        public static void Main() {
                const int low16 = BitConverter.IsLittleEndian ? 0 : 2;
                const int high16 = BitConverter.IsLittleEndian ? 2 : 0;
        }
}
```

Reply:
@NattyNarwhal it looks like .NET Core does, at least I can compile it locally (coreclr with the changes)

Reply:
UPD: no it does not